gk/main.c

```diff
@@ -93,3 +106,22 @@ run_gk(const struct gk_config *gk_conf)
 
 	return 0;
 }
+
+int
+cleanup_gk(struct gk_config *gk_conf)
+{
+	/*
+	 * Atomically decrements the atomic counter (v) by one and returns true 
+	 * if the result is 0, or false in all other cases.
+	 */
+	if (rte_atomic32_dec_and_test(&gk_conf->ref_cnt)) {
+		if (!gk_conf) {
```

Comment:
This `if` statement is unnecessary; just drop it.

If `gk_conf` were `NULL` at this line of code, it would mean that the previous `if` statement would've issued a segment fault.